### PR TITLE
 3496: Vendoring pwasm-utils 0.16 under the name wasm-utils

### PR DESCRIPTION
**Pull request by real-felix**
The `pwasm-utils` crate is not maintained anymore (the repository is archived), so this commit vendors it under the name `wasm-utils`. The code is simplified as well, with the ability to use `no-std` being removed since it is not used in the `execution_engine` crate.

### CODE REVIEW

File: wasm-utils/src/export_globals.rs
Comment (rafal-ch) [inline wasm-utils/src/export_globals.rs:48]: I think we need to update our `Makefile` to include these tests in the `test-rs` job.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
I'd prefer if this weren't added as a new crate, but rather as a module inside EE.

If we do want to stick with making this a crate, we'll need to make a few more changes to support that.

I'll block the PR until we get a decision since it shouldn't merge as is.

**Comment by RitaMAllenCA** [comment]
new name: casper-pwasm-utils

**Comment by real-felix** [comment]
PR no more relevant since it's now forked and lives in its own repo.

**Pull request closed without merging**
